core/modules/filter/filter.module

```diff
@@ -1342,12 +1342,21 @@ function filter_process_format($element) {
       '#parents' => array_merge($element['#parents'], array('format')),
     );
   }
+  // If there's only a single text format available to the current user, then
+  // render it as a hidden form element.
   else {
     $element['format']['format'] = array(
       '#type' => 'hidden',
       '#value' => $element['#format'],
       '#default_value' => $element['#format'],
-      '#attributes' => array('class' => array('filter-list')),
+      '#attributes' => array(
+        'class' => array('filter-list'),
+        // This hidden input element contains only the machine name of the text
+        // format (as its value). We need the human-readable name of the text
+        // format, to be shown as a tooltip/summary in the "Formating options"
+        // fieldset; so we pass that as a data-* attribute.
+        'data-available-editor' => array($options[$element['#format']]),
```

Comment:
I went with this solution here, in order to pass the human-readable name of the text format to the js. Please let me know if using something like the following is preferred instead:

```php
  backdrop_add_js(array('editor-name' => $options[$element['#format']]), 'setting');
```

Reply:
I think this approach is fine (it appears we do this elsewhere in core too), I'd just suggest a different name for the data attribute...

```suggestion
        'data-format-name' => array($options[$element['#format']]),
```

Reply:
Agreed, but better be less vague and use `data-text-format-name`